### PR TITLE
Additions structure field tx changed to txs

### PR DESCRIPTION
**Pull request by Shourya742**
### Description
This PR solves the issue : #954 

### Notes to the reviewers
Additions structure field "tx" is changed to "txs" and relevant changes are made to all files using it.

#### All Submissions:

*  I've signed all my commits
*  I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
*  I ran `cargo fmt` and `cargo clippy` before committing

#### Bugfixes:
* I'm linking the issue being fixed by this PR

### CODE REVIEW

File: crates/chain/src/tx_graph.rs
Comment (vladimirfomene) [inline crates/chain/src/tx_graph.rs]: I think you should also change `txout` to `txouts`

Comment (bit-aloo) [reply crates/chain/src/tx_graph.rs]: ok... On it..

Comment (bit-aloo) [reply crates/chain/src/tx_graph.rs]: Made the changes asked ... Please review @vladimirfomene

**Review by vladimirfomene — CHANGES_REQUESTED** [review]
Can you squash all the commits into one?

**Comment by Shourya742** [comment]
> Can you squash all the commits into one?

I am facing some issues while squashing the commit. Don't know why there are merge conflicts. I am quite new to this and it's a bit confusing. Please help me with this. Sorry for the inconvenience. 
Screenshot (396)

**Comment by vladimirfomene** [comment]
@Shourya742 I think you merged your changes into upstream/master instead of rebasing your changes on it. That is what is stopping you from squashing these changes. You have to do an interactive rebase where you get rid of that merge commit. You can read more about rebase here: https://www.youtube.com/watch?v=7Mh259hfxJg. Let me know how it goes.

**Comment by Shourya742** [comment]
Thank you @vladimirfomene for your assistance, I gained a lot of knowledge from this experience. Could you kindly review the branch and provide any suggestions for changes? I appreciate your help!

**Review by vladimirfomene — APPROVED** [review]
Review ACK 002684891e0bd4225d16ce845bdb7a5b680054f1. You might want to read this article: https://danielabrozzoni.com/posts/contributing_to_oss/. It will help you on your journey as you strive to make progress as a contributor.

**Comment by Shourya742** [comment]
@vladimirfomene Thanks, will go through the article.

**Comment by Shourya742** [comment]
@vladimirfomene Hello, may I inquire if there is any particular issue concerning the pull request? I'm curious as to why it has not yet been merged.

**Comment by vladimirfomene** [comment]
@evanlinjin, can you look into merging this?

**Review by danielabrozzoni — APPROVED** [review]
utACK 002684891e0bd4225d16ce845bdb7a5b680054f1 - looks good to me

**Comment by danielabrozzoni** [comment]
I can't merge as your commits aren't signed, can you fix? https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits

**Comment by evanlinjin** [comment]
Thank you for the work but please don't merge this. Will result in a lot of pain and conflicts imo. 

I suggest we wait until #976 is merged, or we base this work on top of #976.

However, this is such a minor change that can be handled with rust-analyzer in one go anyway.

**Comment by Shourya742** [comment]
Thank you, @evanlinjin, for your assistance. There are no problems at the moment. I will patiently wait until #976 is merged. Once it is, I will update the commit and request a review. I am hopeful that my first contribution will be merged. In the meantime, I will explore other issues.

**Comment by evanlinjin** [comment]
@Shourya742 because #976 is basically ready to be merged (waiting on reviews by others). I suggest that we rebase this work on top of that PR. I can cherry-pick that commit to be part of #976.

Remember that we need a signed commit.

**Comment by Shourya742** [comment]
Hello @evanlinjin, I have rebased my local issue branch onto your wallet_redesign branch. It is possible that I may have misunderstood the instruction to "rebase this work on top of that PR." Please find attached the current commit graph for your reference. Moreover, I have resolved all merge conflicts encountered during the rebase and have made necessary changes to electrum_ext.rs.

<details><summary>Commit Graph</summary>

</details>

**Comment by evanlinjin** [comment]
@Shourya742 where did you push it? I don't have access to your local branches.

The commit I see in this PR is 002684891e0bd4225d16ce845bdb7a5b680054f1 which is different to that in the image.

**Comment by Shourya742** [comment]
@evanlinjin No, I haven't pushed the code yet. I was just hoping to ensure that I handled the rebase correctly or if I may have made any errors. If everything is in order, I would proceed with pushing the code.

**Comment by evanlinjin** [comment]
@Shourya742 the commit history looks good. Please make sure tests pass, clippy is happy and everything compiles before pushing.

**Comment by Shourya742** [comment]
@evanlinjin I have completed running the tests, and Clippy appears to be good. I have also signed the commit. Kindly review it, please.

File: crates/chain/src/chain_graph.rs
Comment (evanlinjin) [inline crates/chain/src/chain_graph.rs:1]: I think you accidentally readded this file.

**Review by evanlinjin — CHANGES_REQUESTED** [review]

File: crates/chain/src/chain_graph.rs
Comment (bit-aloo) [reply crates/chain/src/chain_graph.rs:1]: ok on it!!!

**Comment by Shourya742** [comment]
@evanlinjin have a look...

**Comment by evanlinjin** [comment]
I've cherry-picked the commit to #976 and gave it a proper commit message. In the future, please remember to clean up your commit messages.

The new commit: 965cbe6dde99a1a58168cc0f4bdd320f0aba10e1

This PR is no longer needed.

**Pull request closed without merging**
